dhis-2/dhis-web/dhis-web-api/src/main/java/org/hisp/dhis/webapi/controller/event/RelationshipController.java

```diff
@@ -0,0 +1,310 @@
+    // READ
+    // -------------------------------------------------------------------------
+
+    @GetMapping
+    public List<Relationship> getRelationships(
+        @RequestParam( required = false ) String tei,
+        @RequestParam( required = false ) String enrollment,
+        @RequestParam( required = false ) String event
+    )
+        throws WebMessageException
+    {
+        if ( tei != null )
+        {
+            TrackedEntityInstance trackedEntityInstance = trackedEntityInstanceService.getTrackedEntityInstance( tei );
+
+            if ( trackedEntityInstance != null )
+            {
+                return relationshipService.getRelationshipsByTrackedEntityInstance( trackedEntityInstance, false );
+            }
+            else
+            {
+                throw new WebMessageException( notFound( "No trackedEntityInstance '" + tei + "' found." ) );
+            }
+        }
+        else if ( enrollment != null )
+        {
+            ProgramInstance programInstance = programInstanceService.getProgramInstance( enrollment );
+
+            if ( programInstance != null )
+            {
+                return relationshipService.getRelationshipsByProgramInstance( programInstance, false );
+            }
+            else
+            {
+                throw new WebMessageException( notFound( "No enrollment '" + enrollment + "' found." ) );
+            }
+        }
+        else if ( event != null )
+        {
+            ProgramStageInstance programStageInstance = programStageInstanceService.getProgramStageInstance( event );
+
+            if ( programStageInstance != null )
+            {
+                return relationshipService.getRelationshipsByProgramStageInstance( programStageInstance, false );
+            }
+            else
+            {
+                throw new WebMessageException( notFound( "No event '" + event + "' found." ) );
+            }
+        }
+        else
+        {
+            throw new WebMessageException( badRequest( "Missing required parameter 'tei', 'enrollment' or 'event'." ) );
+        }
+    }
+
+    @GetMapping( "/{id}" )
+    public Relationship getRelationship(
+        @PathVariable String id
+    )
+        throws WebMessageException
+    {
+        Relationship relationship = relationshipService.getRelationshipByUid( id );
+
+        if ( relationship == null )
+        {
+            throw new WebMessageException( notFound( "No relationship with id '" + id + "' was found." ) );
+        }
+
+        return relationship;
+    }
+
+    // -------------------------------------------------------------------------
+    // CREATE
+    // -------------------------------------------------------------------------
+
+    @PostMapping( value = "", consumes = APPLICATION_JSON_VALUE, produces = APPLICATION_JSON_UTF8_VALUE )
+    public WebMessage postRelationshipJson(
+        @RequestParam( defaultValue = "CREATE_AND_UPDATE" ) ImportStrategy strategy,
```

Comment:
Usually for POST we have default strategy set to CREATE

Reply:
And the default strategy should probably be centralized. How to we do this other places?

Reply:
Enrollment and Event has CREATE, but Tei has CREATE_AND_UPDATE. I changed RelationshipController to use CREATE now.